core/lib/workarea/ext/jbuilder/jbuilder_cache.rb

```diff
@@ -0,0 +1,31 @@
+require 'jbuilder/jbuilder_template'
+
+decorate JbuilderTemplate do
+  def _cache_fragment_for(key, options)
+    return yield if workarea_admin?
+
+    super
+  end
+
+  def _cache_key(key, options)
+    super.tap do |result|
+      result << workarea_cache_varies if workarea_cache_varies.present?
+    end
+  end
+
+  private
+
+  def workarea_admin?
+    @context&.controller&.current_user&.admin?
+  rescue ::RuntimeError
```

Comment:
This should only really happen in the storefront API, but we don't know whether API is installed yet so I'm rescuing the base `RuntimeError` class here.